hubstorage/client.py

```diff
@@ -21,22 +42,48 @@ class HubstorageClient(object):
     DEFAULT_TIMEOUT = 60.0
 
     def __init__(self, auth=None, endpoint=None, connection_timeout=None,
-            max_retries=0):
+            max_retries=3):
         self.auth = xauth(auth)
         self.endpoint = endpoint or self.DEFAULT_ENDPOINT
         self.connection_timeout = connection_timeout or self.DEFAULT_TIMEOUT
-        self.session = self._create_session(max_retries)
+        self.session = self._create_session()
+        self.retrier = Retrying(stop_max_attempt_number=max_retries + 1, retry_on_exception=_hc_retry_on_exception)
```

Comment:
woops, my note on `DEFAULT_TIMEOUT ` not being used is wrong, it's used by the resourcestype api (damn you intellij `find usage`). By default, that'll trigger a timeout if haproxy is hanging (requests timeout apply on the server response, and it defaults to 60s)
Sorry about that.